app/controllers/DeductionsFromPaymentsController.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import actions.ActionsProvider
+import config.{AppConfig, ErrorHandler}
+import forms.FormsProvider
+import models.pages.DeductionsFromPaymentsPage
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.SessionHelper
+import views.html.DeductionsFromPaymentsView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class DeductionsFromPaymentsController @Inject()(actionsProvider: ActionsProvider,
+                                                 formsProvider: FormsProvider,
+                                                 pageView: DeductionsFromPaymentsView,
+                                                 errorHandler: ErrorHandler)
+                                                (implicit cc: MessagesControllerComponents, appConfig: AppConfig, ec: ExecutionContext)
+  extends FrontendController(cc) with I18nSupport with SessionHelper {
+
+  def show(taxYear: Int): Action[AnyContent] = actionsProvider.tailoringEnabledFilterWithEndOfYear(taxYear) { implicit request =>
+    Ok(pageView(DeductionsFromPaymentsPage(taxYear, formsProvider.deductionFromPaymentsForm(request.user.isAgent))))
+  }
+
+  def submit(taxYear: Int): Action[AnyContent] = actionsProvider.tailoringEnabledFilterWithEndOfYear(taxYear).async { implicit request =>
+    // TODO: change the redirects once the CYA data has been refactored
+    // the redirects should work as follows:
+    //  1. If user answers 'Yes' and is coming from 'Change' Link and CYA model is complete then redirect to CYA controller.
```

Comment:
For the gateway pages in the personal FE, there's a redirect to a CYA controller. The CYA controllers there don't require a contract and period so the redirects can be made easily (the only parameter required is the tax year). We can't do the redirect easily here as like you said we need contract and period parameters for the CYA controller route.

This issue needs to be discussed between BAs and tech leads to determine what should be done for the CIS frontend.

Reply:
@georgievh 
Since this is just page creation, this should be fine for now. It will need discussing between Devs and BAs on how to best handle this later on.
Likely, the CYA page and models will need refactoring to accommodate this.

Reply:
Thank you @jhkldev, @TheDeadlyPianist